src/bin/background-worker.rs

```diff
@@ -34,9 +34,22 @@ fn main() {
 
     let username = dotenv::var("GIT_HTTP_USER");
     let password = dotenv::var("GIT_HTTP_PWD");
-    let credentials = match (username, password) {
-        (Ok(u), Ok(p)) => Some((u, p)),
-        _ => None,
+    let ssh_key = dotenv::var("GIT_SSH_KEY");
+    let credentials = match (username, password, ssh_key) {
+        (extra_user, extra_pass, Ok(encoded_key)) => {
+            if let (Ok(_), Ok(_)) = (extra_user, extra_pass) {
+                println!("warning: both http and ssh credentials to authenticate with git are set");
+                println!("note:note:  ssh credentials will take precedence over the http ones");
```

Comment:
Did you mean to duplicate this "note:note:" here?

Reply:
Pressed the wrong keybinding probably :sweat_smile: Amended.